[boost-mysql] Add missing dependency on boost-compat

dfederm opened the pull request:
Fixes #44294

dfederm commented:
Hmm, maybe it's `scripts\boost\generate-ports.ps1` that needs updating?

dg0yt commented:
> Hmm, maybe it's `scripts\boost\generate-ports.ps1` that needs updating?

Yes. 

And you will create boost-mysql version 1.87.0 `port-version` 1.

dfederm committed:
[boost-mysql] Add missing dependency on boost-compat

BillyONeal committed:
generate-ports.ps1:

31: Bump port-version of everything edited here
16: Rename scriptsDir to scriptsBoostDir to better indicate that it isn't scripts
86: Reduce the supports expression for boost-cobalt; this is as written in https://github.com/microsoft/vcpkg/pull/42678 which passed PR build, so the less restrictive one seems workable.
381: Add pre-source-stubs and patch-stubs to handle patches like in boost-interprocess et al.

Add stubs to avoid meaningful edits to other portfile.cmakes.

BillyONeal commented:
Sorry for the blizzard of extra changes here; the actual bug that this exposed was that in https://github.com/microsoft/vcpkg/pull/42678 we did not rerun generate-ports.ps1, which detected this missing dependency itself.

BillyONeal commented:
/azp run
